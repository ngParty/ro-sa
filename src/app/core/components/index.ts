import { TitleComponent } from './title/title.component';

export { TitleComponent };
export const COMPONENTS = [TitleComponent];
