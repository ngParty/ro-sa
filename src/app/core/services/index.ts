import { UserService } from './user.service';

export { UserService };
export const PROVIDERS = [UserService];
