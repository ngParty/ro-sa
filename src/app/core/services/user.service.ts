import { Injectable } from '@angular/core';

let nextId = 1;

@Injectable()
export class UserService {
  id = nextId++;
  private _userName = 'Sherlock Holmes';

  get userName() {
    // Demo: add a suffix if this service has been created more than once
    const suffix = this.id > 1 ? ` times ${this.id}` : '';
    return this._userName + suffix;
  }
}
