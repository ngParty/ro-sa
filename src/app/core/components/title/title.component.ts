import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { UserService } from '../../services';

@Component({
  selector: 'app-title',
  template: `
    <h1>
      Welcome to {{ title }}!
    </h1>
  `,
  styleUrls: ['./title.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TitleComponent {
  @Input() title = 'Unknown app';

  user = this.userService.userName;

  constructor(private userService: UserService) {}
}
