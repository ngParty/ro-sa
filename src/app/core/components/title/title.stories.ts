import { text, withKnobs } from '@storybook/addon-knobs/angular';
import { moduleMetadata, storiesOf } from '@storybook/angular';

import { UserService } from '../../services';
import { TitleComponent } from './title.component';

storiesOf(`${TitleComponent.name}`, module)
  .addDecorator(withKnobs)
  .addDecorator(
    moduleMetadata({
      providers: [UserService],
      declarations: [TitleComponent],
    })
  )
  .add('default', () => {
    return {
      template: `<app-title></app-title>`,
    };
  })
  .add('with title input', () => {
    return {
      template: `<app-title [title]="title"></app-title>`,
      props: {
        title: text('title', 'App'),
      },
    };
  });
