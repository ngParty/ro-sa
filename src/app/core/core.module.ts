import { CommonModule } from '@angular/common';
import { ModuleWithProviders, NgModule, Optional, SkipSelf } from '@angular/core';

import { COMPONENTS } from './components';
import { PROVIDERS } from './services';

@NgModule({
  imports: [CommonModule],
  exports: [COMPONENTS],
  declarations: [COMPONENTS],
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error('CoreModule is already loaded. Import it in the AppModule only');
    }
  }

  static forRoot(): ModuleWithProviders {
    return {
      ngModule: CoreModule,
      providers: [PROVIDERS],
    };
  }
}
