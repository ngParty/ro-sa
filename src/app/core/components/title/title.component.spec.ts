import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { UserService } from '../../services';
import { TitleComponent } from './title.component';

describe('TitleComponent', () => {
  let component: TitleComponent;
  let fixture: ComponentFixture<TitleComponent>;

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        declarations: [TitleComponent],
        providers: [UserService],
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(TitleComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });
});
