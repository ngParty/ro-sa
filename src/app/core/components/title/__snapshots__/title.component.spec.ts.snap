// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleComponent should create 1`] = `
<app-title
  title={[Function String]}
  user={[Function String]}
  userService={[Function UserService]}
>
  
      <h1>
    
        Welcome to Unknown app!
      
  </h1>
    
</app-title>
`;
